File: Daily/sbgo/Leetcode1734-解码异或后的排列.go
package sbgo

func decode(encoded []int) []int {
	n := len(encoded) + 1
	ans := make([]int, n)
	x := 0
	for i := 1; i <= n; i++ {
		x ^= i
	}
	cur := 0
	for i := 0; i < n-1; i += 2 {
		cur ^= encoded[i]
	}
	last := cur ^ x
	ans[n-1] = last
	for i := n - 2; i >= 0; i-- {
		ans[i] = ans[i+1] ^ encoded[i]
	}
	return ans
}

File: Daily/PY/Leetcode1734-解码异或后的排列.py
from typing import List
class Solution:
    def decode(self, encoded: List[int]) -> List[int]:
        n = len(encoded) + 1
        ans = [0] * n
        x = 0
        for i in range(1,n + 1):
            x ^= i
        cur = 0
        for i in range(0,n - 1,2):
            cur ^= encoded[i]
        last = cur ^ x
        ans[n - 1] = last
        for i in range(n-2,-1,-1):
            ans[i] = ans[i+1] ^ encoded[i]
        return ans